runtime/JavaScript/src/antlr4/context/ParserRuleContext.d.ts

```diff
@@ -11,6 +11,7 @@ export declare class ParserRuleContext extends RuleContext {
     parentCtx: ParserRuleContext | undefined;
     exception?: RecognitionException;
     parser?: Parser;
+    ruleIndex: number;
```

Comment:
this breaks the build, and there is no ruleIndex defined in ParserRuleContext.js

Reply:
In DefaultErrorStrategy.js, in reportFailedPredicate method, at line 295, I see this code:
`
const ruleName = recognizer.ruleNames[recognizer._ctx.ruleIndex]
`
As far as my understanding, recognizer in this case is a Parser instance, _ctx is a ParserRuleContext. Therefore, ruleIndex should be available in ParserRuleContext.

Reply:
That works in js because in the above context, the _ctx is always a subclass of ParserRuleContext for which ruleIndex is defined. But there are subclasses for which it is not defined. See the ParserRuleContext code, there is no ruleIndex. Exposing one would be misleading, and it breaks the build.

Reply:
If you need to write typescript code that mimics DefaultErrorStrategy, you can create an IndexedRuleContext interface with a ruleContext field, and cast the _ctx.